src/index.tsx

```diff
@@ -46,26 +56,154 @@ function Updaters() {
+    false
+  )
+
+  useEffect(() => {
+    const injectedState: ConnectorState = {
+      isActive: injectedIsActive,
+      previousIsActive: previousInjectedIsActive,
+      isActivating: injectedIsActivating,
+      previousIsActivating: previousInjectedIsActivating,
+      isEagerlyConnecting: isInjectedEagerlyConnecting,
+      setIsEagerlyConnecting: setIsInjectedEagerlyConnecting,
+    }
+    const coinbaseWalletState: ConnectorState = {
+      isActive: coinbaseWalletIsActive,
+      previousIsActive: previousCoinbaseWalletIsActive,
+      isActivating: coinbaseWalletIsActivating,
+      previousIsActivating: previousCoinbaseWalletIsActivating,
+      isEagerlyConnecting: isCoinbaseWalletEagerlyConnecting,
+      setIsEagerlyConnecting: setIsCoinbaseWalletEagerlyConnecting,
+    }
+    const walletConnectState: ConnectorState = {
+      isActive: walletConnectIsActive,
+      previousIsActive: previousWalletConnectIsActive,
+      isActivating: walletConnectIsActivating,
+      previousIsActivating: previousWalletConnectIsActivating,
+      isEagerlyConnecting: isWalletConnectEagerlyConnecting,
+      setIsEagerlyConnecting: setIsWalletConnectEagerlyConnecting,
+    }
+    const isActiveMap = new Map<Wallet, ConnectorState>([
+      [Wallet.INJECTED, injectedState],
+      [Wallet.COINBASE_WALLET, coinbaseWalletState],
+      [Wallet.WALLET_CONNECT, walletConnectState],
+    ])
+
+    isActiveMap.forEach((state: ConnectorState, wallet: Wallet) => {
+      const {
+        isActive,
+        previousIsActive,
+        isActivating,
+        previousIsActivating,
+        isEagerlyConnecting,
+        setIsEagerlyConnecting,
+      } = state
+
+      if (!isActivating && previousIsActivating) {
+        setIsEagerlyConnecting(false)
+      }
+
+      if (isActive === false && previousIsActive === undefined && isActivating) {
+        setIsEagerlyConnecting(true)
+      }
+
+      if (isActive && previousIsActive === false) {
+        if (isEagerlyConnecting) {
+          setIsEagerlyConnecting(false)
+          if (!walletOverrideBackfilled) {
+            dispatch(setWalletOverride({ wallet }))
+          }
+        } else {
+          dispatch(setWalletOverride({ wallet }))
+        }
+      }
+    })
+  }, [
+    dispatch,
+    walletOverrideBackfilled,
+    injectedIsActive,
+    coinbaseWalletIsActive,
+    walletConnectIsActive,
+    previousInjectedIsActive,
+    previousCoinbaseWalletIsActive,
+    previousWalletConnectIsActive,
+    injectedIsActivating,
+    coinbaseWalletIsActivating,
+    walletConnectIsActivating,
+    previousInjectedIsActivating,
+    previousCoinbaseWalletIsActivating,
+    previousWalletConnectIsActivating,
+    isInjectedEagerlyConnecting,
+    isCoinbaseWalletEagerlyConnecting,
+    isWalletConnectEagerlyConnecting,
+    setIsInjectedEagerlyConnecting,
+    setIsCoinbaseWalletEagerlyConnecting,
+    setIsWalletConnectEagerlyConnecting,
+  ])
+
+  return (
+    <Blocklist>
```

Comment:
This should be passed in as props.children and then rendered, otherwise it couples all of the app state together.